Drift/Private/DriftBase.cpp

```diff
@@ -2928,25 +2928,39 @@ void FDriftBase::InitServerRootInfo()
         }
     }
 
-    auto request = GetRootRequestManager()->Get(drift_url);
-    request->OnResponse.BindLambda([this, drift_url](ResponseContext& context, JsonDocument& doc)
+    const auto Request = GetRootRequestManager()->Get(DriftURL);
+    Request->OnResponse.BindLambda([this](ResponseContext& Context, JsonDocument& Doc)
     {
-        if (!JsonArchive::LoadObject(doc[TEXT("endpoints")], driftEndpoints))
+        if (!JsonArchive::LoadObject(Doc[TEXT("endpoints")], driftEndpoints))
         {
-            context.error = TEXT("Failed to parse drift endpoints");
+            Context.error = TEXT("Failed to parse drift endpoints");
             state_ = DriftSessionState::Disconnected;
             return;
         }
+
         InitServerAuthentication();
         eventManager->SetEventsUrl(driftEndpoints.eventlogs);
         onStaticRoutesInitialized.Broadcast();
     });
-    request->OnError.BindLambda([this](ResponseContext& context)
+    Request->OnError.BindLambda([this](ResponseContext& Context)
     {
-        context.errorHandled = true;
-        Reset();
+        Context.errorHandled = true;
+
+    	DRIFT_LOG(Base, Error, TEXT("Failed to fetch Drift endpoints"));
+
+    	if (Context.response.IsValid() && Context.response->GetResponseCode() >= 500)
+    	{
+    		DRIFT_LOG(Base, Log, TEXT("Retrying fetching Drift endpoints"));
+
+    		// Retry if server error (5XX HTTP status code)
+    		InitServerRootInfo();
```

Comment:
Generally, you don't want to just retry blindly over and over. Add a random back-off, and limit retries to N, where N is commonly 3. The heartbeat implements this pattern already.

Reply:
With your recent addition to adding random back off into the Drift retry system and me now using it with a set amount of retries, I assume this comment is resolved.

Reply:
Yes